src/leo-client-app/src/components/Scheduler.tsx

```diff
@@ -1,165 +1,478 @@
+        setScheduleCommands((prevCommands) => ({
+          ...prevCommands,
+          [scheduleId]: data.commands ?? [],
+        }));
+      })
+      .catch((error) => {
+        console.error("Error fetching schedule commands:", error);
+      });
   };
 
   useEffect(() => {
-    fetchValidCommands(satelliteId);
+    fetchSchedules(satelliteId);
   }, [satelliteId]);
 
-  // Mutation function
-  const { mutate } = useMutation({
-    mutationFn: () =>
-      sendCommandSchedule(userId, scheduleId, satelliteId, currentSchedule),
-
-    onSuccess: () => {
-      queryClient.invalidateQueries({ queryKey: ["useGetCommandsBySchedule"] });
-      setCurrentSchedule([]);
-    },
-
-    onSettled: () => {
-      queryClient.invalidateQueries({ queryKey: ["useGetCommandsBySchedule"] });
-      setCurrentSchedule([]);
-    },
-  });
-
-  // Function will load schedule somewhere, currently console log for POC demo
-  const sendSchedule = async () => {
-    mutate();
-  };
+  const router = useRouter();
 
   return (
-    <div
-      style={{
-        display: "flex",
-        flexDirection: "column",
-        alignItems: "flex-start",
-        gap: "4rem",
-      }}>
-      <div
-        style={{
-          display: "flex",
-          justifyContent: "space-around",
-          alignItems: "flex-start",
-        }}>
-        <div
-          style={{
-            minWidth: "200px",
-            border: "2px solid white",
-            borderRadius: "16px",
-            padding: "10px",
-          }}>
-          <h2>Valid Commands</h2>
-          <div>
-            {validCommands &&
-              validCommands.length > 0 &&
-              validCommands.map((command, index) => (
-                <button
-                  key={index}
-                  className="scheduleButton"
-                  onClick={() => addCommand(command)}>
-                  {command}
-                </button>
+    <div className="upcomingSchedulesBox" >
+      <p className="headerBox">Satellite Name</p>
+      <p className="headerBox2">All Schedules</p>
+      <p className="headerBox3">Schedule Queue</p>
+
+      <div className="main-schedule"> 
+
+      <Stack alignItems="flex-start" spacing={1}>
+        {isLoading ? (
+          <Box className="loadingBox">
+            <CircularProgress />
+          </Box>
+        ) : (
+          <Grid
+            className="futureSchedulesBox"
+            container
+            spacing={2}
+            sx={{
+              display: "flex",
+              // flexWrap: "nowrap",
+              // overflowX: "auto",
+              maxWidth: "10vw", // Use 100vw to ensure it considers the full viewport width
+              boxSizing: "border-box",
+              "& .MuiGrid-item": {
+                flex: "0 0 auto",
+              },
+              mx: -2,
+            }}
+          >
+            {schedules1 &&
+              schedules1.map((schedule, index) => (
+                <Grid item key={index}>
+                  <Card
```

Comment:
i know these are copied over from my cards, but these cards are inherently different, so they should be formatted and treated differently (mainly sizing)